File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[][] map1 = arrToMap(n, arr1);
        String[][] map2 = arrToMap(n, arr2);
        String[] answer = new String[n];
        for (int i = 0; i < n; i++) {
            StringBuilder sb = new StringBuilder();
            for (int j = 0; j < n; j++) {
                if (!map1[i][j].equals(" ") || !map2[i][j].equals(" ")) {
                    sb.append("#");
                } else {
                    sb.append(" ");
                }
            }
            answer[i] = sb.toString();
        }
        return answer;
    }
    
    public static String[][] arrToMap(int n, int[] arr) {
        String[][] map = new String[n][n];
        String[] biArr = new String[n];
        for (int i = 0; i < n; i++) {
            String biStr = Integer.toBinaryString(arr[i]);
            StringBuilder sb = new StringBuilder();
            if (biStr.length() < n) {
                sb.append("0".repeat(n - biStr.length()));
                sb.append(biStr);
                biArr[i] = sb.toString();
            } else {
                biArr[i] = biStr;
            }
        }
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (Character.toString(biArr[i].charAt(j)).equals("0")) {
                    map[i][j] = " ";
                } else {
                    map[i][j] = "#";
                }
            }
        }
        return map;
    }
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.swift
func solution(_ n:Int, _ arr1:[Int], _ arr2:[Int]) -> [String] {
//     func makeMap(_ arr: [Int], _ n: Int) -> [[String]] {
//         var map = Array(repeating: Array(repeating: " ", count: n), count: n)

//         var convertArr = arr.map { String(format: "%0\(n)ld", Int(String($0, radix: 2))!) }
//         for i in 0..<n {
//             for j in 0..<n {
//                 if Array(convertArr[i])[j] == "1" {
//                     map[i][j] = "#"
//                 }
//             }
//         }

//         return map
//     }

//     func result(_ map1: [[String]], _ map2: [[String]]) -> [String] {
//         var map = Array(repeating: Array(repeating: " ", count: map1.count), count: map1.count)

//         for i in 0..<n {
//             for j in 0..<n {
//                 if map1[i][j] == "#" || map2[i][j] == "#" {
//                     map[i][j] = "#"
//                 }
//             }
//         }

//         return map.map { $0.joined() }
//     }

//     return result(makeMap(arr1, n), makeMap(arr2, n))
    return (0..<n).map {
        let binary = String(arr1[$0] | arr2[$0], radix: 2)
        let padded = String(repeating: "0", count: n - binary.count) + binary
        return padded.reduce("") { $0 + ($1 == "0" ? " " : "#") }
    }
}